src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs

```diff
@@ -144,7 +144,12 @@ public override void Initialize(AnalysisContext analysisContext)
                 }
 
                 // Compute code metrics.
-                var computeTask = CodeAnalysisMetricData.ComputeAsync(compilationContext.Compilation, compilationContext.CancellationToken);
+                var inheritanceExcludedTypes = compilationContext.Options.GetInheritanceExcludedSymbolNamesOption(CA1501Rule, compilationContext.Compilation,
+                    compilationContext.CancellationToken);
+
+                var metricsAnalysisContext = new CodeMetricsAnalysisContext(compilationContext.Compilation, compilationContext.CancellationToken,
+                    namedType => inheritanceExcludedTypes.Contains(namedType));
+                var computeTask = CodeAnalysisMetricData.ComputeAsync(metricsAnalysisContext);
```

Comment:
Can you open a tracking issue to update the command line Metrics tool to also respect this option: https://github.com/dotnet/roslyn-analyzers/blob/d9c93107393265b20a982c7445184a57e3c47003/src/Tools/Metrics/Program.cs#L319

At the minimum, we should update it to exclude System.* types. Please also make a note that we should change the internal Visual Studio code base for "Analyzer -> Calculate Code Metrics" command to also have similar change so this is consistent everywhere.

Reply:
See #3275